### PR TITLE
General overhaul

### PR DESCRIPTION
**Pull request by Photon-GitHub**
First of all this pr adds maven as a buildtool to easily use this repository.
Moreover it adds a new dragon spawn command to manually respawn the dragon with timed delay if desired. 
Lastly it also removes support for minecraft 1.13.0 as in my opinion most people should have updated to 1.13.1/1.13.2 by now and adding a new dependency to maven is not worth it.

### CODE REVIEW

**Comment by Photon-GitHub** [comment]
I also increased the version number to 1.6.0

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        </repository>
+    </repositories>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.spigotmc</groupId>
+            <artifactId>spigot-api</artifactId>
+            <version>1.13.2-R0.1-SNAPSHOT</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (2008Choco) [inline pom.xml]: In inclusion of the API is unnecessary if the server is depended on below as the server shades the API. This dependency can be removed.

File: src/main/java/com/ninjaguild/dragoneggdrop/commands/DragonSpawnCmd.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import org.bukkit.command.TabCompleter;
+import org.bukkit.entity.EnderCrystal;
+import org.bukkit.entity.EnderDragon;
+import org.bukkit.entity.Player;
+
+import java.util.Collections;
+import java.util.List;
+
+public class DragonSpawnCmd implements CommandExecutor, TabCompleter
+{
```
Comment (2008Choco) [inline src/main/java/com/ninjaguild/dragoneggdrop/commands/DragonSpawnCmd.java]: Try to stay true to the programming style of this repository (K&R bracket indentation, tabs vs spaces, etc.)

File: src/main/java/com/ninjaguild/dragoneggdrop/commands/DragonSpawnCmd.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            return true;
+
+        plugin.getDEDManager().getWorldWrapper(world).startRespawn(DEDManager.RespawnType.COMMAND);
+        return true;
+    }
+
+    @Override
+    public List<String> onTabComplete(CommandSender sender, Command command, String alias, String[] args) {
+        return Collections.emptyList();
+    }
```
Comment (2008Choco) [inline src/main/java/com/ninjaguild/dragoneggdrop/commands/DragonSpawnCmd.java]: If all that's returned is an empty list, TabCompleter need not be overridden. This can be removed.

File: src/main/java/com/ninjaguild/dragoneggdrop/commands/DragonTemplateCmd.java [hunk 4]
```diff
@@ -17,6 +12,10 @@
 import org.bukkit.entity.Player;
 import org.bukkit.inventory.ItemStack;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
```
Comment (2008Choco) [inline src/main/java/com/ninjaguild/dragoneggdrop/commands/DragonTemplateCmd.java:18]: Unnecessary changes to import order, though the change to NumberUtils (lang3 vs lang) is fine.

**Review by 2008Choco — CHANGES_REQUESTED** [review]
Overall, the changes are fine with some minor notes (see below). General gist of the requested changes are as follows:
- Tabs should be used instead of spaces, as well as K&R bracket indentation style to keep consistent with the project
- Dependency on the API is unnecessary
- Redundant implementation of TabCompleter in the command
- Change in import order should be omitted from the PR

The rest seems alright. I had intended to make changes to excessive whitespace in the future to clean up the project, and I likely will take another review in the near future, but I will do so after this PR has been cleaned up a bit and merged. :) I appreciate your contribution thus far.

File: src/main/java/com/ninjaguild/dragoneggdrop/commands/DragonTemplateCmd.java (hunk 4 above)
Comment (Photon-GitHub) [reply src/main/java/com/ninjaguild/dragoneggdrop/commands/DragonTemplateCmd.java:18]: That is the standard way IntelliJ sorts it. I don't really see a reason to change this especially as the previous version also doesn't have an apparent pattern for me.

**Comment by Photon-GitHub** [comment]
@2008Choco Changed everything except the imports which I deem ok.

File: .idea/codeStyles/Project.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<component name="ProjectCodeStyleConfiguration">
+  <code_scheme name="Project" version="173">
+    <codeStyleSettings language="JAVA">
+      <indentOptions>
+        <option name="USE_TAB_CHARACTER" value="true" />
+      </indentOptions>
+    </codeStyleSettings>
+  </code_scheme>
+</component>
```
Comment (2008Choco) [inline .idea/codeStyles/Project.xml]: IDE-specific file. I suppose this isn't in the gitignore, nor is much else for IntelliJ. Remove this file and I'll update the .gitignore to be much more inclusive of IDE files after this is merged

**Review by 2008Choco — APPROVED** [review]
